### PR TITLE
Adding cors support #2

### LINKED ISSUE #2: Adding CORS Support
I think we should add CORS support adding a BIG UPPERCASE WARNING Indicating to narrow down the CORS statement.

```bash
npm install cors -P
```

```bash
npm install @types/cors -P
```

```typescript
import cors from 'cors';

// ...

// Warning: narrow down your CORS allowed origins to the ones your cliente applications //options for cors midddleware
const options: cors.CorsOptions = {
  allowedHeaders: [
    'Origin',
    'X-Requested-With',
    'Content-Type',
    'Accept',
    'X-Access-Token',
  ],
  credentials: true,
  methods: 'GET,HEAD,OPTIONS,PUT,PATCH,POST,DELETE',
  // IMPORTANT LIMIT HERE YOUR CLIENT APPS DOMAINS
  origin: '*',
  preflightContinue: false,
};

app.use(cors(options));
```

https://brianflove.com/2017-03-22/express-cors-typescript/

### PR DESCRIPTION
**Pull request by brauliodiez**
This fixed #2 we are adding guidance on how to configure CORS (just added on the readme.md, not adding it to the real codebase since it could imply securirty risks if a given developer just uses it out of the box without reading the guide).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,68 @@ npm run start:debug
...
+
+_/src/app.ts_
+
+```diff
+import { createApp } from './express.server';
++ import cors from 'cors';
+import { envConstants } from './env.constants';
+import { api } from './api';
+
+const app = createApp();
```
Comment (nasdan) [inline README.md:58]: I would move this config to `./express.server` to keep express server config together, what do you think?:

_/src/express.server.ts_

```diff
import express from 'express';
import bodyParser from 'body-parser';
+ import cors from 'cors';

+ // Warning: narrow down your CORS allowed origins to the web application domains that
+ // are allowed to access this application.
+ // check the methods and origin properties in the CorsOptions
+ const options: cors.CorsOptions = {
+  allowedHeaders: [
+    'Origin',
+    'X-Requested-With',
+    'Content-Type',
+    'Accept',
+    'X-Access-Token',
+  ],
+  credentials: true,
+  // IMPORTANT YOU MAY LIMIT THE ALLOWED VERBS TO FOR INSTANCE ONLY GET
+  methods: 'GET,HEAD,OPTIONS,PUT,PATCH,POST,DELETE',
+  // IMPORTANT LIMIT THIS HERE TO YOUR CLIENT APPS DOMAINS
+  origin: '*',
+  preflightContinue: false,
+ };

export const createApp = () => {
  const app = express();

  app.use(bodyParser.urlencoded({ extended: false }));
  app.use(bodyParser.json());
+ app.use(cors(options));

  return app;
};

```

File: README.md [hunk 2]
```diff
@@ -24,6 +24,68 @@ npm run start:debug
...
++    'Content-Type',
++    'Accept',
++    'X-Access-Token',
++  ],
++  credentials: true,
++  // IMPORTANT YOU MAY LIMIT THE ALLOWED VERBS TO FOR INSTANCE ONLY GET
++  methods: 'GET,HEAD,OPTIONS,PUT,PATCH,POST,DELETE',
++  // IMPORTANT LIMIT THIS HERE TO YOUR CLIENT APPS DOMAINS
++  origin: '*',
++  preflightContinue: false,
```
Comment (nasdan) [inline README.md:76]: Is this property required?

File: README.md (hunk 1 above)
Comment (brauliodiez) [reply README.md:58]: Cool, adding @delamux to the loop.

@delamux I can give you write acces to this repo and you can work on this branch to add the enhancements, are you ok with that?

File: README.md (hunk 2 above)
Comment (brauliodiez) [reply README.md:76]: Not sure, something we have to check

Comment (delamux) [reply README.md:76]: Is added by default in configuration in the cors repo. here is explained: https://github.com/expressjs/cors/issues/153
if I understood right, is too keep the cors in the middleware when you jump with next() method.

**Comment by delamux** [comment]
Upsss! this one you had assigned @brauliodiez , so sorry I forgot. In mine PR I added in `express.server.ts`

**Pull request merged**